ppl/build.gradle

```diff
@@ -59,6 +59,7 @@ dependencies {
     testImplementation group: 'org.hamcrest', name: 'hamcrest-library', version: '2.1'
     testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.12.4'
 
+    api group: 'org.opensearch', name: 'opensearch', version: "${opensearch_version}"
```

Comment:
PPL module should not depend on org.opensearch. I think there are 2 options.  (1) Move PPLQueryRequest to plugin module or (2) Create TransportPPLQueryRequest if required

Reply:
1. Why ppl module shouldn't depend on org.opensearch?
2. option 1 will introduce circular dependency, due to 
   1. `PPLService`(in ppl project) uses `PPLQueryRequest`(in plugin project)
   2. `createPPLService`(in plugin project) uses `PPLService`(in ppl project)
3. I will probably go for option 3, but I still don't think that's a good option. Because we are maintaining duplicate code. The fields of `TransportPPLQueryRequest` and `PPLQueryRequest` will be mostly the same. Therefore, I still think if allowed, we should add opensearch as dependency of ppl module

Reply:
We try to make PPL/SQL as a query engine which not bind to OpenSearch. OpenSearch is just on storage, we working on add others, e.g. Prometheus. 
* TransportPPLQueryRequest is used between Rest layer and Transport layer which align with opensearch framework.
* PPLQueryRequest is the interface required by PPLService which is not bind to opensearch.

For example, we can run PPL as a library and use memory as storage. In this case, all the Rest and Transport related stuff is exclude, we just create a PPLService.

Reply:
Got it, that makes sense to me. I'll update the PR with specific `TransportPPLQueryRequest` model